homeassistant/components/systemmonitor/coordinator.py

```diff
@@ -2,27 +2,55 @@
 
 from __future__ import annotations
 
-from abc import abstractmethod
+from dataclasses import dataclass
 from datetime import datetime
 import logging
 import os
-from typing import NamedTuple, TypeVar
+from typing import Any, NamedTuple
 
 from psutil import Process
 from psutil._common import sdiskusage, shwtemp, snetio, snicaddr, sswap
 import psutil_home_assistant as ha_psutil
 
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_component import DEFAULT_SCAN_INTERVAL
-from homeassistant.helpers.update_coordinator import (
-    TimestampDataUpdateCoordinator,
-    UpdateFailed,
-)
+from homeassistant.helpers.update_coordinator import TimestampDataUpdateCoordinator
 from homeassistant.util import dt as dt_util
 
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True, kw_only=True)
+class SensorData:
+    """Sensor data."""
+
+    disk_usage: dict[str, sdiskusage]
+    swap: sswap
+    memory: VirtualMemory
+    io_counters: dict[str, snetio]
+    addresses: dict[str, list[snicaddr]]
+    load: tuple[float, float, float]
+    cpu_percent: float | None
+    boot_time: datetime
+    processes: list[Process]
+    temperatures: dict[str, list[shwtemp]]
+
+    def as_dict(self) -> dict[str, Any]:
```

Comment:
I was going to suggest https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict but I guess you want to stringily them for diagnostics

Reply:
Yes. I tested with asdict but as it can't serialize it properly and I prefer to have a "nice" output I went with this.